src/Microsoft.AspNetCore.Razor.Language/RazorProjectEngineBuilderExtensions.cs

```diff
@@ -4,11 +4,75 @@
 using System;
 using System.Linq;
 using Microsoft.AspNetCore.Razor.Language.CodeGeneration;
+using Microsoft.AspNetCore.Razor.Language.Intermediate;
 
 namespace Microsoft.AspNetCore.Razor.Language
 {
     public static class RazorProjectEngineBuilderExtensions
     {
+        /// <summary>
+        /// Registers a class configuration delegate that gets invoked during code generation.
+        /// </summary>
+        /// <param name="builder">The <see cref="RazorProjectEngineBuilder"/>.</param>
+        /// <param name="configureClass"><see cref="Action"/> invoked to configure 
+        /// <see cref="ClassDeclarationIntermediateNode"/> during code generation.</param>
+        /// <returns>The <see cref="RazorProjectEngineBuilder"/>.</returns>
+        public static RazorProjectEngineBuilder ConfigureClass(
+            this RazorProjectEngineBuilder builder,
+            Action<RazorCodeDocument, 
+            ClassDeclarationIntermediateNode> configureClass)
```

Comment:
-100 PranavPoints

Reply:
I have restored Taylor's pranavpoints, and his honour.